### DISCUSSION TITLE
Can we add a `delay` argument to `invalidateQueries` after a mutation?

### DISCUSSION THREAD

**Discussion by githorse** [TanStack/query#5859] [Ideas]
Due to some questionable backend design I'm not able to change, sometimes I need to wait a few seconds after performing a mutation before the updated data is available to the query. How about adding a delay as an optional argument to `invalidateQueries`? In this non-ideal world it's not crazy that an update might not be immediate after a mutation, so this seems like it might be a useful feature.

**Comment by TkDodo** [comment]
> How about adding a delay as an optional argument to invalidateQueries?

no, sorry. you can debounce the call to `invalidateQueries` on your end.
